### PR TITLE
[Feature](ThreadPool) Support thread pool per disk for scanners

### LINKED ISSUE #7581: [Roadmap] Implement IO manager
root issue

### PR DESCRIPTION
**Pull request by yinzhijian**
# Proposed changes

Issue Number: close #7581

## Problem Summary:

Support thread pool per disk for scanners to prevent pool performance from some high ioutil disks happening

key point:
1. each disk has a thread pool for scanners
2. whenever a thread pool of one disk runs out of local work, tasks can be retrieved from other threads(disks). This is done round-robin.

performance testing: 
vec version: 25% faster than single thread pool in a high io util disk test case
normal version: 8% faster than single thread pool in a high io util disk test case

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No)
3. Has document been added or modified: (No)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/priority_work_stealing_thread_pool.hpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    std::vector<std::shared_ptr<BlockingPriorityQueue<Task>>> _work_queues;
+
+    // Collection of worker threads that process work from the queues.
+    ThreadGroup _threads;
+
+    // Guards _empty_cv
+    std::mutex _lock;
+
+    // Set to true when threads should stop doing work and terminate.
+    std::atomic<bool> _shutdown;
```
Comment (morningman) [inline be/src/util/priority_work_stealing_thread_pool.hpp:154]: Use `_shutdown` in parent class

File: be/src/util/priority_thread_pool.hpp [hunk 2]
```diff
@@ -39,6 +39,7 @@ class PriorityThreadPool {
     public:
         int priority;
         WorkFunction work_function;
+        int queue_id;
```
[the remark this answers is not in the document]
Comment (yinzhijian) [reply be/src/util/priority_thread_pool.hpp:42]: task_id is not mean which disk this task belongs to, so queue_id maybe better.

[the remark this answers is not in the document]
Comment (morningman) [reply be/src/util/priority_thread_pool.hpp:42]: Oh, i read it as `query_id`...
lgtm

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**